Feat/render highight md

nickperry12 opened the pull request:
## Description

Added a preview button in the working memory section of the playground. This button opens a modal, rendering the markdown, allowing users to preview the current content or any changes they're making.

<img width="728" height="246" alt="image" src="https://github.com/user-attachments/assets/ef4bd560-e61b-4c98-bde4-f3e7539e0a86" />

<img width="468" height="381" alt="image" src="https://github.com/user-attachments/assets/c68c1def-91ba-4d9e-a67a-7cd4d104a762" />


## Related Issue(s)

Fixes #5959 

## Type of Change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Code refactoring
- [ ] Performance improvement
- [ ] Test update

## Checklist

- [ ] I have made corresponding changes to the documentation (if applicable)
- [ ] I have added tests that prove my fix is effective or that my feature works

nickperry12 committed:
Added a preview feature, allowing users to preview working memory markdown

- Added a preview button
- Button opens a modal allowing users to preview their rendered markdown
- Modal adjusts to the size of the markdown in both width and height

nickperry12 committed:
Slight refactor

- small refactor on how the event listeners are handled for the escape key

nickperry12 commented on packages/cli/src/playground/src/domains/agents/markdown-preview-dialog.tsx:
```diff
@@ -0,0 +1,57 @@
+    };
+
+    if (open) {
+      document.addEventListener('keydown', handleEscape, true);
```
@greptileai The Dialog's built-in escape handling didn't seem to be working with the modal. Couldn't identify why that was the case, so I added this in to provide that functionality

nickperry12 replied:
@greptileai got it. Change made, and committing now 👍

nickperry12 committed:
Small refactor

Small refactor to make conditional rendering of the button more consistent with current patterns.

Using the bubbling phase instead of the capture phase for the event listener on the modal

DanielSLew committed:
Merge branch 'main' into feat/render-highight-md

TylerBarnes reviewed (changes requested):
This is awesome, thank you! Only thing that doesn't make sense to me is having a "preview" button. This is the only place this text will ever be rendered, so previewing isn't the thing to do. By default it should just show the current "preview" as what you see initially. Then when you press edit it turns back into an editable raw text box

nickperry12 committed:
Refactor/

- Removed the modal and preview button, and instead render the markdown in the text area
- Ensured consistency with the scrollbar display with markdown and structured schema
- Verified that structured memory display remains the same

Co-Authored-By: Jason Jones <jasonjabarjones@gmail.com>

nickperry12 committed:
Merge branch 'feat/render-highight-md' of https://github.com/nickperry12/mastra into feat/render-highight-md

nickperry12 commented:
@TylerBarnes made all the changes and now have it rendered in the text area when not editing. There was a bit of fighting with the CSS to get the scroll bars matching with the markdown and structured memory, but finally got it. Also ensured it appears the same in both Chrome and Firefox.

<img width="1385" height="723" alt="image" src="https://github.com/user-attachments/assets/d2b2e5f1-9b20-4bab-a232-e2600c6d97a3" />

<img width="1402" height="725" alt="image" src="https://github.com/user-attachments/assets/270ffbe9-2b40-4afc-aabc-314fc31c6a93" />

TylerBarnes reviewed (approved):
nice!!

nickperry12 committed:
chore

Fixed lint issue

Co-Authored-By: Jason Jones <jasonjabarjones@gmail.com>

abhiaiyer91 committed:
Merge branch 'main' into feat/render-highight-md

TylerBarnes committed:
Merge branch 'main' into feat/render-highight-md
